src/Aspire.Hosting/ApplicationModel/BeforeResourceStartedEvent.cs

```diff
@@ -0,0 +1,25 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Aspire.Hosting.Eventing;
+
+namespace Aspire.Hosting.ApplicationModel;
+
+/// <summary>
+/// This event is raised by orchestrators before they have started a new resource.
+/// </summary>
+/// <param name="resource">The resource that is being created.</param>
+/// <param name="services">The <see cref="IServiceProvider"/> for the app host.</param>
+/// <remarks>
+/// Resources that are created by orchestrators may not yet be ready to handle requests.
+/// </remarks>
+public class BeforeResourceStartedEvent(IResource resource, IServiceProvider services) : IDistributedApplicationResourceEvent
```

Comment:
We typically use the words "-ing" and "-ed" for before and after events. Examples:

https://github.com/dotnet/runtime/blob/3622bfa4329faa73551b2508aade7777a04a3ca8/src/libraries/Microsoft.Extensions.Hosting.Abstractions/src/IHostApplicationLifetime.cs#L22-L28

https://learn.microsoft.com/en-us/dotnet/standard/design-guidelines/names-of-type-members#names-of-events

✔️ DO give events names with a concept of before and after, using the present and past tenses.

For example, a close event that is raised before a window is closed would be called Closing, and one that is raised after the window is closed would be called Closed.

❌ DO NOT use "Before" or "After" prefixes or postfixes to indicate pre- and post-events. Use present and past tenses as just described.

Reply:
@davidfowl should I just revert my commit :)